adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDDFunctions.scala

```diff
@@ -49,6 +51,65 @@ class NucleotideContigFragmentRDDFunctions(rdd: RDD[NucleotideContigFragment]) e
   }
 
   /**
+   * Save nucleotide contig fragments in FASTA format.
+   *
+   * @param fileName file name
+   */
+  def adamSaveAsFasta(fileName: String) = {
+
+    def isFragment(record: NucleotideContigFragment): Boolean = {
+      (Option(record.fragmentNumber), Option(record.numberOfFragmentsInContig)) match {
+        case (None, None)       => false
+        case (None, Some(m))    => false
+        case (Some(n), None)    => false
+        case (Some(n), Some(m)) => (m > 1)
+      }
+    }
+
+    def toFasta(record: NucleotideContigFragment): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(record.contig.contigName)
+      Option(record.description).foreach(n => sb.append(" ").append(n))
+      if (isFragment(record)) {
+        sb.append(" fragment %d of %d".format(record.fragmentNumber + 1, record.numberOfFragmentsInContig))
+      }
+      for (line <- Splitter.fixedLength(60).split(record.fragmentSequence)) {
+        sb.append("\n")
+        sb.append(line)
+      }
+      sb.toString
+    }
+
+    rdd.map(toFasta).coalesce(1).saveAsTextFile(fileName)
+  }
+
+  /**
+   * Merge fragments by contig name.
+   */
+  def mergeFragments(): RDD[NucleotideContigFragment] = {
+
+    def merge(first: NucleotideContigFragment, second: NucleotideContigFragment): NucleotideContigFragment = {
+      val merged = NucleotideContigFragment.newBuilder()
+        .setContig(first.contig)
+        .setDescription(first.description)
+        .setFragmentNumber(null)
+        .setFragmentStartPosition(null)
+        .setNumberOfFragmentsInContig(null)
+        .setFragmentSequence(first.fragmentSequence + second.fragmentSequence)
+        .build
+
+      merged
+    }
+
+    rdd
+      .sortBy(fragment => (fragment.contig.contigName, Option(fragment.fragmentNumber) match { case Some(n) => n.toInt case None => -1 }))
+      .map(fragment => (fragment.contig, fragment))
+      .reduceByKey(merge)
```

Comment:
Ah, I think I see why you're doing the sort first. Are you trying to ensure that the `reduceByKey` runs in sorted order? If so, I'm not _sure_ that this approach is guaranteed to work. Technically, the guaranteed correct way would be to do:

```
rdd.map(f => (f.getContig.getContigName, (f.getFragmentNumber, f)))
  .groupByKey()
  .map(kv => {
    val (_, records) = kv
    records.sortBy(_._1)
  .map(_._2)
  .reduce(merge)
  })
```